File: codingame/easy/temperatures/temperatures.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <iterator>
#include <sstream>


using namespace std;

int main()
{
    int n; // the number of temperatures to analyse
    cin >> n; cin.ignore();
    std::string input; // the n temperatures expressed as integers ranging from -273 to 5526
    getline(cin, input);

    int closest = 5527;
    int tst;
    
    std::vector<int> inputs;
    std::istringstream in( input );
    std:copy( std::istream_iterator<int> (in), std::istream_iterator<int>(),
        std::back_inserter( inputs ));
        
    for(int i : inputs) {
        if ( abs(closest) == abs(i) and closest < 0 ) {
            closest = i;
        } else if (abs(closest) > abs(i) ) {
            closest = i;
        }
    
    }

    if (closest == 5527 ) {
        closest = 0;
    }
    
    cout << closest << endl;
}

File: codingame/easy/temperatures/temperatures.java
import java.util.*;
import java.io.*;
import java.math.*;

class Solution {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt(); // the number of temperatures to analyse
        
        if (in.hasNextLine()) {
            in.nextLine();
        }
        
        String line = in.nextLine();
        
        //System.out.println(line);
        
           // System.out.println(line.length());
        
        String[] stemps = line.split(" ");
        int[] temps = new int[stemps.length];
        int closest = 99999;
        int tst;        
        //System.out.println(stemps[0]);
      
        if (line.length() > 0) {
                
            for (int i = 0; i < stemps.length; i++) {
                //System.out.println(stemps[i]);
                
               temps[i] = Integer.parseInt(stemps[i]); 
            }  
            
            closest = temps[0];
     
            for(int i = 1; i <= temps.length -1; i++)
            {
               tst = temps[i];
               //System.out.println(tst);
               //System.out.println(closest);
               
               if (Math.abs(closest) == Math.abs(tst) && closest < 0)
                    closest = temps[i];
                else if (Math.abs(closest) > Math.abs(tst))
                    closest = temps[i];
               }       
            
        }

        if (closest == 99999) {
            closest = 0;   
        }
        //System.out.println("------");

        System.out.println(closest);
        //System.out.println("#######");
    }
}
